10 get by topic

ShaunRainey opened the pull request:
Completes up to task 10. Will add user functionality soon

nc-christian-dutton commented on src/Utilities/api.js:
```diff
@@ -44,4 +44,11 @@ export const postComment = (article_id, newCommentText, user) => {
+    return ncNews.get(`articles?topic=${topic}`)
+    .then((response) => {
+        return response.data.articles
+    })
```
Instead of creating a new api call, you can reuse your existing getArticles function. Have a look at axios queries. You can give a get request an object containing a query. If it's there it will use it, if not it will return all of the articles. 

If you implement this, you might be able to clean up your useEffect too!

nc-christian-dutton commented:
Good implementation, I've left a refactoring suggestion above...but it works! 

The loading gif is great by the way 😄

ShaunRainey commented:
Cheers Christian. I've added this function now and removed a block from the
use effect statement. I'll look at adding the feedback from the previous
pull request into the submission for error handling.

I know it's really simple code and it took me about 2 minutes, but that
loading feature is my proudest part of this project 😅



On Thu, 20 Jun 2024 at 12:23, nc-christian-dutton ***@***.***>
wrote:

> Good implementation, I've left a refactoring suggestion above...but it
> works!
>
> The loading gif is great by the way 😄
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ShaunRainey/nc-news/pull/2#issuecomment-2180435864>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/BEESANKT3WEE4QATRAGH6Y3ZIK3S3AVCNFSM6AAAAABJTPSXAKVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDCOBQGQZTKOBWGQ>
> .
> You are receiving this because you modified the open/close state.Message
> ID: ***@***.***>
>
